dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/TrackerIdentifierCollector.java

```diff
@@ -199,4 +200,13 @@ private <T> void addIdentifier(
 
     identifiers.computeIfAbsent(klass, k -> new HashSet<>()).add(identifier);
   }
+
+  private <T> void addIdentifier(
+      Map<Class<?>, Set<String>> identifiers, Class<T> klass, UID identifier) {
```

Comment:
`identifier` cannot be null as this is guaranteed by the `BodyConverter`, correct? Can we add some `@Nonnull` on fields to make that explicit?

Reply:
if that's the case, should we remove the null check?

Reply:
`BodyConverter` is guaranteeing that the `UID` of an entity is going to be populated but we are using `UID`s also for related entities so an `Event` can have `enrollment` field `null` and it is a `UID`.
So I still need to check if the identifier is `null` here